crates/polars-io/src/ipc/ipc_stream.rs

```diff
@@ -359,6 +359,18 @@ impl<W: Write> IpcStreamWriter<W> {
     }
 }
 
+impl<W: Write + Seek> IpcStreamWriter<W> {
+    pub fn finish_with_stream_position(mut self, df: &mut DataFrame) -> PolarsResult<usize> {
+        self.finish(df)?;
+
+        if let Ok(pos) = self.writer.stream_position() {
+            return Ok(pos as usize);
+        } else {
+            polars_bail!(ComputeError: "Can't get position in writer")
```

Comment:
I think the behaviour should be the same for `IpcStreamBatchedWriter` and `IpcStreamWriter`, either both bail if `stream_position` fails or both fallback to bytes written.

Reply:
Done